File: test2.py
def is_matching(expression):
    stack = []  # 用于存储左括号的栈

    # 定义左右括号的映射关系
    brackets_map = {')': '(', ']': '[', '}': '{'}

    for char in expression:
        if char in '({[':
            # 如果是左括号，入栈
            stack.append(char)
        elif char in ')}]':
            # 如果是右括号
            if not stack or stack[-1] != brackets_map[char]:
                # 如果栈为空或者栈顶的左括号与当前右括号不匹配，返回0
                return 0
            stack.pop()  # 匹配成功，弹出左括号

    # 如果栈为空，说明括号全部匹配
    return int(not stack)

# 输入算术表达式
expression = input("请输入算术表达式: ")

# 输出结果
result = is_matching(expression)
print(result)

File: test2.cpp
#include <iostream>
#include <vector>
#include <stack>

int isStackSequence(int inputLength, const std::vector<int>& inputSequence, const std::vector<int>& outputSequence) {
    std::stack<int> st;
    int outputIndex = 0;
    int popCount = 0;

    for (int num : inputSequence) {
        st.push(num);

        while (!st.empty() && outputIndex < inputLength && st.top() == outputSequence[outputIndex]) {
            st.pop();
            outputIndex++;
            popCount++;
        }
    }

    return (outputIndex == inputLength) ? popCount : 0;
}

int main() {
    int inputLength;
    // std::cout << "Enter the length of the input sequence: ";
    std::cin >> inputLength;

    std::vector<int> inputSequence(inputLength);
    // std::cout << "Enter the input sequence: ";
    for (int i = 0; i < inputLength; ++i) {
        std::cin >> inputSequence[i];
    }

    std::vector<int> outputSequence(inputLength);
    // std::cout << "Enter the output sequence: ";
    for (int i = 0; i < inputLength; ++i) {
        std::cin >> outputSequence[i];
    }

    int result = isStackSequence(inputLength, inputSequence, outputSequence);

    std::cout << result << std::endl;
    system("pause");
    return 0;
}
